store/snapshots/manager.go

```diff
@@ -363,7 +366,20 @@ func (m *Manager) doRestoreSnapshot(snapshot types.Snapshot, chChunks <-chan io.
 		return payload.Payload, nil
 	}
 
-	nextItem, err = m.multistore.Restore(snapshot.Height, snapshot.Format, streamReader)
+	// chStorage is the channel to pass the KV pairs to the storage snapshotter.
+	chStorage := make(chan *store.KVPair, defaultStorageChannelBufferSize)
+	defer close(chStorage)
+
+	storageErrs := make(chan error, 1)
+	go func() {
+		err := m.storageSnapshotter.Restore(snapshot.Height, chStorage)
+		if err != nil {
+			storageErrs <- err
+		}
+		close(storageErrs)
```

Comment:
I highly recommend instead making this a defer right at the top so that even if .Restore panics, storageErrs will always be closed. Also a naming nit to make code easier to read, for channels, just suffix it with Ch so `storageErrsCh`